packages/react/src/components/searchInput/SearchInput.test.tsx

```diff
@@ -71,6 +71,36 @@ describe('<SearchInput /> spec', () => {
     expect(onSubmit.mock.calls[0][0]).toBe('Apple');
   });
 
+  it('submits the selected value on enter press or icon click', async () => {
+    const onSubmit = jest.fn();
+    const { getAllByLabelText, container } = render(
+      <SearchInput<SuggestionItemType>
+        label="search"
+        suggestionLabelField="value"
+        getSuggestions={getSuggestions}
+        onSubmit={onSubmit}
+      />,
+    );
+
+    const input = getAllByLabelText('search')[0];
+    userEvent.type(input, 'abc{enter}');
+    await waitFor(() => {
+      expect(onSubmit.mock.calls[0][0]).toBe('abc');
+    });
+    const buttons = container.querySelectorAll('button');
+    const clearButton = buttons[0];
```

Comment:
Could this work in here:
const input = screen.getByLabelText('Clear', { selector: 'button' })

Reply:
Works